depends/packages/boost.mk

```diff
@@ -26,6 +27,7 @@ $(package)_cxxflags_freebsd=-fPIC
 endef
 
 define $(package)_preprocess_cmds
+  patch -p1 < $($(package)_patch_dir)/darwin.diff
```

Comment:
If this were made conditional on macOS, then it wouldn't break other platforms in the case of a problem with applying the diff. For example, something like:

```
ifeq ($(host_os),darwin)
$(package)_patches=darwin.diff

define $(package)_preprocess_cmds
  patch -p1 < $($(package)_patch_dir)/darwin.diff
endef
endif
```

(and delete the unconditional `$(package)_patches=darwin.diff` above).

Reply:
Thete is no need for conditional as this file is only applicable on Darwin.

Reply:
In any case, the patch is super safe because it’s applied on a ghost file. The patched file is not in the repo. It doesn’t exist before the compilation and gets deleted after the compilation :)